src/test/lib/MtuTest.cpp

```diff
@@ -0,0 +1,256 @@
+        TEST_QUIC_SUCCEEDED(Listener.GetLocalAddr(ServerLocalAddr));
+
+        const uint16_t MinimumMtu = 1248;
+
+        {
+            MsQuicSettings Settings;
+            Settings.SetMaximumMtu(1450).SetMinimumMtu(MinimumMtu);
+
+            MsQuicCredentialConfig ClientCredConfig;
+            MsQuicConfiguration ClientConfiguration(Registration, Alpn, Settings, ClientCredConfig);
+            TEST_TRUE(ClientConfiguration.IsValid());
+
+            MsQuicConnection Connection(Registration);
+            TEST_QUIC_SUCCEEDED(Connection.GetInitStatus());
+            TEST_QUIC_SUCCEEDED(Connection.StartLocalhost(ClientConfiguration, ServerLocalAddr));
+            TEST_NOT_EQUAL(nullptr, Context.Connection);
+
+            //
+            // Set connection settings after open, should fail
+            //
+            Settings.SetMaximumMtu(1400).SetMinimumMtu(1300);
+            TEST_QUIC_STATUS(
+                QUIC_STATUS_INVALID_PARAMETER,
+                MsQuic->SetParam(
+                    Connection.Handle,
+                    QUIC_PARAM_LEVEL_CONNECTION,
+                    QUIC_PARAM_CONN_SETTINGS,
+                    sizeof(Settings),
+                    &Settings));
+
+            MsQuicSettings CheckSettings;
+            uint32_t CheckSize = sizeof(CheckSettings);
+            TEST_QUIC_SUCCEEDED(
+                MsQuic->GetParam(
+                    Connection.Handle,
+                    QUIC_PARAM_LEVEL_CONNECTION,
+                    QUIC_PARAM_CONN_SETTINGS,
+                    &CheckSize,
+                    &CheckSettings));
+
+            Connection.Shutdown(1);
+
+            TEST_EQUAL(1450, CheckSettings.MaximumMtu);
+            TEST_EQUAL(MinimumMtu, CheckSettings.MinimumMtu);
+        }
+    }
+}
+
+static
+QUIC_STATISTICS
+GetConnStatistics(_In_ MsQuicConnection& Conn) {
+    QUIC_STATISTICS value = {};
+    uint32_t valueSize = sizeof(value);
+    QUIC_STATUS Status =
+        MsQuic->GetParam(
+            Conn.Handle,
+            QUIC_PARAM_LEVEL_CONNECTION,
+            QUIC_PARAM_CONN_STATISTICS,
+            &valueSize,
+            &value);
+    if (QUIC_FAILED(Status)) {
+        TEST_FAILURE("MsQuic->GetParam(CONN_STATISTICS) failed, 0x%x.", Status);
+    }
+    return value;
+}
+
+void
+QuicMtuDiscoveryTest(
+    _In_ int Family,
+    _In_ BOOLEAN DropClientProbePackets,
+    _In_ BOOLEAN DropServerProbePackets,
+    _In_ BOOLEAN RaiseMinimumMtu
+    )
+{
+    MsQuicRegistration Registration;
+    TEST_TRUE(Registration.IsValid());
+
+    const uint16_t MinimumMtu = RaiseMinimumMtu ? 1360 : 1280;
+    const uint16_t MaximumMtu = 1500;
+
+    MsQuicAlpn Alpn("MsQuicTest");
+    MsQuicSettings Settings;
+    Settings.SetMinimumMtu(MinimumMtu).SetMaximumMtu(MaximumMtu);
+
+    MsQuicConfiguration ServerConfiguration(Registration, Alpn, Settings, ServerSelfSignedCredConfig);
+    TEST_TRUE(ServerConfiguration.IsValid());
```

Comment:
I know you copied this from my (bad) example, but it's better to do the following:
```c
TEST_QUIC_SUCCEEDED(ServerConfiguration.GetInitStatus());
```
So we get the actual error on failure.